### DISCUSSION TITLE
How to Configure phpredis to Handle Redis Master Failover with Sentinel?

### DISCUSSION THREAD

**Discussion by irtaza9** [phpredis/phpredis#2504] [Q&A] [answered]
I've set up Redis with high availability (HA) on three machines using Sentinel. The configuration is working well, and I have the following IP addresses for my machines:

172.29.22.8
172.16.179.96
172.29.28.17

Each Redis instance has its own IP address and they are interconnected. When the master node goes down, one of the slaves is promoted to master.

I am using phpredis to connect to the Redis server. However, I'm unsure how to handle the connection in my application to ensure it connects to the current master, even when failover occurs. How can I configure phpredis to handle this situation?

Additionally, if there are any best practices or configurations I should be aware of for managing this with phpredis and Redis Sentinel, please let me know.

**Comment by yatsukhnenko** [comment, marked as answer]
Currently PhpRedis has only basic support of sentinel commands and doesn't provide any build-in functionality to handle master-slave failover automatically. For now you need to implement failover logic by yourself in PHP code

**Reply by irtaza9** [reply]
Thank you, right now I'm going with [predis](https://github.com/predis/predis) as this client is providing the support for [sentinel](https://github.com/predis/predis/blob/v2.x/examples/replication_sentinel.php). Anyone in the future required any help regarding the connection issue please ping on this thread.
